New Club Penguin not supported ruffle player

### Describe the bug


https://user-images.githubusercontent.com/80902993/196750362-eb8831bc-c234-41c8-aa41-40d783b669d0.mp4

it's not working, because this is multi files, this is actionsupport 3.0
i know ruffle player multiplayer is not working

### Expected behavior

![Screenshot 2022-10-19 192944](https://user-images.githubusercontent.com/80902993/196750531-8067cbf3-45b9-403d-88e4-334f636bac67.jpg)
you know ''Download'' and ''Launch NewCP App'' it's can work with flash player, his server it's working

### Affected platform

Browser's extension

### Operating system

Windows 11

### Browser

Edge

### Additional information

from https://newcp.net/en/

It still has problem with some language related stuff
```
ERROR core/src/avm2/events.rs:424 Error dispatching event EventObject(EventObject { type: "addedToStage", class: flash.events::Event, ptr: 0x127b638 }) to handler FunctionObject(FunctionObject { ptr: 0x127b130 }) : Error: MainContext could not determine the locale for the specified langauge. Make sure that 'null' has been added to the switch.
```

--- a/web/packages/core/src/ruffle-imports.ts
+++ b/web/packages/core/src/ruffle-imports.ts
@@ -65,14 +65,13 @@ export function copyToAudioBufferInterleaved(
  *
  * @internal
  */
-// @ts-expect-error defined but not used
-// eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/no-unused-vars
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 export function callExternalInterface(name: string, args: any[]): any {
-    // [NA] Yes, this is direct eval. Yes, this is a Bad Thing when it comes to security.
+    // [NA] Yes, this is indirect eval. Yes, this is a Bad Thing when it comes to security.
     // In fact, yes this is vulnerable to an XSS attack!
     // But plot twist: Flash allowed for this and many games *rely on it*. :(
     // Flash content can do `call("eval", "....")` regardless, this doesn't enable anything that wasn't already permitted.
     // It just goes against what the documentation says, and *looks* really suspicious.
     // Content can only run this if the website has enabled `allowScriptAccess`, so it has to be enabled by the website too.
-    return eval(`(${name})(...args)`);
+    return new Function(`return (${name})(...arguments);`)(...args);
 }

--- a/web/packages/selfhosted/test/integration_tests/external_interface/test.ts
+++ b/web/packages/selfhosted/test/integration_tests/external_interface/test.ts
@@ -275,4 +275,26 @@ log called with 1 argument
 `,
         );
     });
+
+    it("doesn't enforce Strict Mode", async () => {
+        const player = await browser.$("<ruffle-object>");
+        await browser.execute((player) => {
+            player.callMethodWithDelay(
+                "function(){return aPropertyThatDoesntExist = 'success!'}",
+            );
+        }, player);
+
+        // [NA] Because of the delay, if we fetch immediately we *may* just get part of the log.
+        await browser.pause(200);
+
+        const actualOutput = await getTraceOutput(browser, player);
+        expect(actualOutput).to.eql(
+            `callMethodWithDelay called with 1 argument
+  [
+    "function(){return aPropertyThatDoesntExist = 'success!'}"
+  ]
+  call(function(){return aPropertyThatDoesntExist = 'success!'}, ...) = "success!"
+`,
+        );
+    });
 });

